src/test/java/org/opensearch/sdk/TestNamedWriteableRegistryApi.java

```diff
@@ -0,0 +1,171 @@
+
+import org.opensearch.Version;
+import org.opensearch.cluster.node.DiscoveryNode;
+import org.opensearch.common.io.stream.NamedWriteable;
+import org.opensearch.common.io.stream.NamedWriteableRegistry;
+import org.opensearch.common.io.stream.NamedWriteableRegistryParseRequest;
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.NamedWriteableRegistryResponse;
+import org.opensearch.common.io.stream.OutputStreamStreamOutput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.common.transport.TransportAddress;
+import org.opensearch.test.OpenSearchTestCase;
+import org.opensearch.extensions.BooleanResponse;
+import org.opensearch.extensions.OpenSearchRequest;
+import org.opensearch.extensions.ExtensionsOrchestrator.OpenSearchRequestType;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class TestNamedWriteableRegistryApi extends OpenSearchTestCase {
+    private List<NamedWriteableRegistry.Entry> namedWriteables;
+    private NamedWriteableRegistryApi namedWriteableRegistryApi;
+
+    private static class Example implements NamedWriteable {
+        public static final String INVALID_NAME = "invalid_name";
+        public static final String NAME = "example";
+        private final String message;
+
+        Example(String message) {
+            this.message = message;
+        }
+
+        Example(StreamInput in) throws IOException {
+            this.message = in.readString();
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            out.writeString(message);
+        }
+
+        @Override
+        public String getWriteableName() {
+            return NAME;
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (o == null || getClass() != o.getClass()) {
+                return false;
+            }
+            Example that = (Example) o;
+            return Objects.equals(message, that.message);
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(message);
+        }
+    }
+
+    @BeforeEach
+    public void setUp() throws Exception {
+        this.namedWriteables = Collections.singletonList(new NamedWriteableRegistry.Entry(Example.class, Example.NAME, Example::new));
+        this.namedWriteableRegistryApi = new NamedWriteableRegistryApi(namedWriteables);
+    }
+
+    @Test
+    public void testNamedWriteableRegistryCreation() {
+        assert (namedWriteableRegistryApi.getRegistry() instanceof NamedWriteableRegistry);
+    }
+
+    @Test
+    public void testNamedWriteableRegistryRequest() throws UnknownHostException {
+        DiscoveryNode sourceNode = new DiscoveryNode(
```

Comment:
Where are we using sourceNode?

Reply:
Residual, forgot to remove. Ill go ahead and make this change